securedrop_client/gui/conversation/export/device.py

```diff
@@ -129,13 +115,60 @@ def print_file(self, file_uuid: str) -> None:
         if not self._controller.downloaded_file_exists(file):
             return
 
-        if not self._controller.qubes:
-            return
-
         self.print_requested.emit([file_location])
 
     def _move_export_service_to_thread(self, thread: QThread) -> None:
         self._export_service_thread = thread
 
         self._export_service.moveToThread(self._export_service_thread)
         self._export_service_thread.start()
+
+
+class StubDevice(QObject):
+    """Abstracts an export service for use in GUI components.
```

Comment:
In practice, is the `StubDevice` used in a production environment, or just during testing?

Reply:
Only in testing environment (we don't have non-Qubes OS production environments). I commented more on this in  https://github.com/freedomofpress/securedrop-client/pull/1541#issuecomment-1208653531 : )